packages/shared/lib/wallet.ts

```diff
@@ -2060,3 +2058,27 @@ export const hasPendingTransactions = (account: WalletAccount): boolean => {
 
     return account?.messages.some((m) => !m.confirmed)
 }
+
+/**
+ * Determines whether an account has any valid pending transactions i.e. transactions that can confirm.
+ *
+ * @method hasValidPendingTransactions
+ *
+ * @param {WalletAccount} account
+ *
+ * @returns {boolean}
+ */
+export const hasValidPendingTransactions = (account: WalletAccount): boolean => {
+    if (!account) return false
+    const pendingMessages = account?.messages.filter((m) => !m.confirmed)
+    const pendingInputs = pendingMessages.flatMap((msg) => {
+        if (msg.payload?.type === 'Transaction') {
+            return msg.payload?.data?.essence?.data?.inputs
+        }
+    })
+    const unspentOutputs = account?.addresses.filter((a) => a.balance > 0).flatMap((a) => Object.values(a.outputs))
+
+    return pendingInputs.some((i) =>
+        unspentOutputs.map((o) => o.transactionId).includes(i.data?.metadata?.transactionId)
```

Comment:
I think `unspentOutputs.some((o) => o.transactionId === i.data?.metadata?.transactionId)` might be slightly more efficient.